task01/script.js

```diff
@@ -0,0 +1,43 @@
+/*Heather Sleyster
+sleyster_assignment05b
+Thoendel
+INFO 2134
+4/9/2020
+*/
+
+//This is my event listener for the load event.
+window.addEventListener('load', function(){
+
+/*These are my variables for myForm id, firstName id, lastName id, telephoneNumber id
+and pattern to check values entered.
+*/
+    const myForm = document.getElementById('myForm');
+    const firstName = document.getElementById('firstName');
+    const lastName = document.getElementById('lastName');
+    const pattern = /^(?:\([2-9]\d{2}\)\?|[2-9]\d{2}(?:\-?|\?))[2-9]\d{2}[- ]?\d{4}$/;
+    const telephoneNumber = document.getElementById('telephoneNumber');
+    const errors = [];
+    /*This is my event listener on the myForm for the submit event. It checks the values 
+    entered for firstName, lastName and telephoneNumber for valid values. If blank fields
+    for firstName or lastName an error alert is displayd and the form cannot submit. If
+    invalid values entered for telephoneNumber are entered an error alert is displayd and
+    the form cannot submit.
+    */
+    myForm.addEventListener('submit',function(event){
+        if (firstName.value == ""){
+            errors.push(alert('Error: First Name cannot be blank'));
+        }
+        if (lastName.value == ""){
+            errors.push(alert('Error: Last Name cannot be blank'));
+        }
+        if (telephoneNumber != pattern){
```

Comment:
this is not how you test with a regex pattern. That is why I always get the message that the phone number is invalid - even when it is valid. To test use something like:

pattern.test(telephoneNumber.value)